src/foam/dao/jdbc/AbstractJDBCDAO.java

```diff
@@ -0,0 +1,283 @@
+//      prop.set(obj, resultSet.getObject(index++));
+    }
+
+    return obj;
+  }
+
+  /**
+   * Sets the value of the PrepareStatement
+   * @param stmt statement to set values
+   * @param obj object to get values from
+   * @return the updated index
+   * @throws SQLException
+   */
+  public void setStatementValues(IndexedPreparedStatement stmt, FObject obj) throws SQLException {
+    Iterator i = properties_.iterator();
+    while ( i.hasNext() ) {
+      PropertyInfo prop = (PropertyInfo) i.next();
+      prop.setStatementValue(stmt, obj);
+    }
+  }
+
+  /**
+   * Closes resources without throwing exceptions
+   * @param resultSet ResultSet
+   * @param stmt IndexedPreparedStatement
+   */
+  public void closeAllQuietly(ResultSet resultSet, IndexedPreparedStatement stmt) {
+    if ( resultSet != null )
+      try { resultSet.close(); } catch (Throwable ignored) {}
+    if ( stmt != null )
+      try { stmt.close(); } catch (Throwable ignored) {}
+  }
+
+  /**
+   * Prepare the formatted column names. Appends column names like: (c1,c2,c3)
+   * @param builder builder to append to
+   */
+  public void buildFormattedColumnNames(FObject obj, StringBuilder builder) {
+    // collect columns list into comma delimited string
+    builder.append("(");
+    Iterator i = properties_.iterator();
+    while ( i.hasNext() ) {
+      PropertyInfo prop = (PropertyInfo) i.next();
+/*       if ( "id".equals(prop.getName()) )
+        continue; */
+
+      builder.append(prop.createStatement());
+      if ( i.hasNext() ) {
+        builder.append(",");
+      }
+    }
+    builder.append(")");
+  }
+
+  /**
+   * Prepare the formatted value placeholders. Appends value placeholders like: (?,?,?)
+   * @param builder builder to append to
+   */
+  public void buildFormattedColumnPlaceholders(FObject obj, StringBuilder builder) {
+    // map columns into ? and collect into comma delimited string
+    builder.append("(");
+    Iterator i = properties_.iterator();
+    while ( i.hasNext() ) {
+      PropertyInfo prop = (PropertyInfo) i.next();
+/*       if ( "id".equals(prop.getName()) )
+        continue; */
+
+      builder.append("?");
+      if ( i.hasNext() ) {
+        builder.append(",");
+      }
+    }
+    builder.append(")");
+  }
+
+  /**
+   * Prepare the formatted UPDATE string like :  description='etc', name='etc'
+   * @param builder builder to append to
+   */
+  public void buildUpdateFormattedColumnNames(FObject obj, StringBuilder builder) {
+    // collect columns list into comma delimited string
+    Iterator i = properties_.iterator();
+    while ( i.hasNext() ) {
+
+      PropertyInfo prop = (PropertyInfo) i.next();
+      if ( "id".equals(prop.getName()) )
+        continue;
+
+      builder.append(prop.createStatement());
+      builder.append("='");
+      builder.append(prop.get(obj));  //add the new property value
+      builder.append("'");
+      if ( i.hasNext() ) {
+        builder.append(",");
```

Comment:
When adding a single character, I think it is faster to just append a character. Ex.: append(',');

Reply:
Currently I do a builder.append("c") to append a character. Is there a faster way?